### PR TITLE
[zh-cn]: update the translation of the try-catch statement

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

As the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#18266

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 1]
```diff
@@ -1,118 +1,100 @@
...
 title: try...catch
 slug: Web/JavaScript/Reference/Statements/try...catch
+l10n:
+  sourceCommit: 4e947e81afc753bedcaaba75f262a07b92511849
 ---
 
 {{jsSidebar("Statements")}}
 
-**`try...catch`** 语句标记要尝试的语句块，并指定一个出现异常时抛出的响应。
+**`try...catch`** 语句由一个 `try` 块和 `catch` 块、`finally` 块或两者组成。首先执行 `try` 块中的代码，如果它抛出异常，则将执行 `catch` 块中的代码。`finally` 块中的代码将在控制流退出整个结构之前始终被执行。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:10]: 第一句再修改一下，现在的翻译不太能体现英文文档的那个意思

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 2]
```diff
@@ -1,118 +1,100 @@
...
-- `condition_1`
-  - : 一个条件表达式。
-- `finally_statements`
-  - : 在`try`语句块之后执行的语句块。无论是否有异常抛出或捕获这些语句都将执行。
+```
+
+- `tryStatements`
+  - : 要执行的语句。
+- `catchStatements`
+  - : 如果 `try` 块抛出异常时执行的语句。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:29]:
```suggestion
  - : `try` 块抛出异常后执行的语句。
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 3]
```diff
@@ -1,118 +1,100 @@
...
-你也可以用`try`语句去处理 JavaScript 异常。参考[JavaScript 指南](/zh-CN/docs/Web/JavaScript/Guide)了解更多关于 Javascript 异常的信息。
+你可以嵌套多个 `try` 语句。如果内部 `try` 语句没有 `catch` 块，则使用包裹它的 `try` 语句的 `catch` 块。
 
-### 无条件的 `catch` 块
+你可以使用 `try` 语句来处理 JavaScript 异常。有关 JavaScript 异常的更多信息，请参阅 [JavaScript 指南](/zh-CN/docs/Web/JavaScript/Guide/Control_flow_and_error_handling#异常处理语句)。
 
-当使用单个无条件`catch`子句时，抛出的任何异常时都会进入到`catch`块。例如，当在下面的代码中发生异常时，控制转移到`catch`子句。
+### catch 绑定
+
+当 `try` 块中抛出异常时，`exceptionVar`（即 `catch (e)`）保存了异常的值。你可以使用这个{{Glossary("binding", "绑定")}}获取有关抛出的异常的信息。这个{{Glossary("binding", "绑定")}}只能在 `catch` 块的{{Glossary("Scope", "范围")}}中使用。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:60]:
```suggestion
当 `try` 块中抛出异常时，`exceptionVar`（即 `catch (e)`）保存了异常的值。你可以使用这个{{Glossary("binding", "绑定")}}获取有关抛出的异常的信息。这个{{Glossary("binding", "绑定")}}只能在 `catch` 块的{{Glossary("Scope", "作用域")}}中使用。
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 4]
```diff
@@ -1,118 +1,100 @@
...
+} catch ({ name, message }) {
+  console.log(name); // “TypeError”
+  console.log(message); // “哦豁”
 }
 ```
 
-`catch`块指定一个标识符（在上面的示例中为 e），该标识符保存由`throw`语句指定的值。`catch`块是唯一的，因为当输入`catch`块时，JavaScript 会创建此标识符，并将其添加到当前作用域；标识符仅在`catch`块执行时存在；`catch`块执行完成后，标识符不再可用。
-
-### 条件 `catch` 块
+`catch` 子句创建的绑定与 `catch` 块处于同一范围内，因此 `catch` 块中声明的变量不能与 `catch` 子句创建的绑定具有相同的名称。（[有一个例外](/zh-CN/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features#语句)，但它是废弃的语法。）
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:73]:
```suggestion
`catch` 子句创建的绑定与 `catch` 块处于同一作用域内，因此 `catch` 块中声明的变量不能与 `catch` 子句创建的绑定具有相同的名称。（[有一个例外](/zh-CN/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features#语句)，但它是已弃用的语法。）
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 5]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
 
-- `try` 块正常执行结束后（没有抛出异常）立即执行；
-- 在 `catch` 块正常执行完毕后立即执行；
-- 紧接着在 `try` 代码块或 `catch` 代码块中的控制流语句（`return`、`throw`、`break`、`continue`）执行之前执行。
+- 在 `try` 块正常执行完成（且没有抛出异常）后立即执行；
+- 在 `catch` 块正常执行完成后立即执行；
+- 在 `try` 块或 `catch` 块中将要执行控制流语句（`return`、`throw`、`break`、`continue`）退出块之前立即执行。
 
-如果 `try` 代码块抛出异常，即使没有 `catch` 代码块来处理异常，`finally` 代码块仍会执行，在这种情况下，异常仍会在 `finally` 代码块执行完毕后立即抛出。
+如果 `try` 块中抛出异常，即使没有 `catch` 块来处理异常，`finally` 块仍然执行，在这种情况下，异常仍然会在`finally` 块执行完成后立即抛出。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:118]:
```suggestion
如果 `try` 块中抛出异常，即使没有 `catch` 块来处理异常，`finally` 块仍然执行，在这种情况下，异常仍然会在 `finally` 块执行完成后立即抛出。
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 6]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
+### 无条件捕获块
+
+当使用 `catch` 块时，`catch` 块将在 `try` 块中抛出异常时被执行。例如，在下面的代码中，控制流将被转移到 `catch` 块。
+
+```js
+try {
+  throw "我的异常"; // 产生异常
+} catch (e) {
+  // 处理任何异常的语句
+  logMyErrors(e); // 将异常对象传递给错误处理程序
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:159]:
```suggestion
  logMyErrors(e); // 将异常对象传递给错误处理器
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 7]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
+}
+```
+
+`catch` 块指定了一个标识符（如上例中的 `e`），它保存了异常的值。这个值只能在 `catch` 块的{{Glossary("Scope", "范围")}}内使用。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:163]:
```suggestion
`catch` 块指定了一个标识符（如上例中的 `e`），它保存了异常的值。这个值只能在 `catch` 块的{{Glossary("Scope", "作用域")}}内使用。
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 8]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
+} catch (e) {
+  if (e instanceof TypeError) {
+    // 处理 TypeError 异常的语句
+  } else if (e instanceof RangeError) {
+    // 处理 RangeError 异常的语句
+  } else if (e instanceof EvalError) {
+    // 处理 EvalError 异常的语句
+  } else {
+    // 处理未指定异常的语句
+    logMyErrors(e); // 将异常对象传递给错误处理程序
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:181]:
```suggestion
    logMyErrors(e); // 将异常对象传递给错误处理器
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 9]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
 ```js
 try {
   try {
-    throw new Error("oops");
+    throw new Error("哦豁");
   } catch (ex) {
-    console.error("inner", ex.message);
+    console.error("内层", ex.message);
   } finally {
     console.log("finally");
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:240]:
```suggestion
    console.log("最终");
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 10]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
-  console.error("outer", ex.message);
+  console.error("外层", ex.message);
 }
 
-// Output:
-// "inner" "oops"
-// "finally"
+// 输出：
+// “内层” “哦豁”
+// “finally”
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:248]:
```suggestion
// “最终”
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 11]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
 try {
   try {
-    throw new Error("oops");
+    throw new Error("哦豁");
   } catch (ex) {
-    console.error("inner", ex.message);
+    console.error("内层", ex.message);
     throw ex;
   } finally {
     console.log("finally");
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:261]:
```suggestion
    console.log("最终");
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 12]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
+  console.error("外层", ex.message);
 }
 
-// Output:
-// "inner" "oops"
-// "finally"
-// "outer" "oops"
+// 输出：
+// “内层” “哦豁”
+// “finally”
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:269]:
```suggestion
// “最终”
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 13]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
+(() => {
   try {
-    throw new Error("oops");
+    try {
+      throw new Error("哦豁");
+    } catch (ex) {
+      console.error("内层", ex.message);
+      throw ex;
+    } finally {
+      console.log("finally");
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:288]:
```suggestion
      console.log("最终");
```

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 14]
```diff
@@ -125,120 +107,198 @@ function isValidJSON(text) {
...
-}
+})();
 
-// 注：此 try catch 语句需要在 function 中运行才能作为函数的返回值，否则直接运行会报语法错误
-// Output:
-// "inner" "oops"
-// "finally"
+// 输出：
+// “内层” “哦豁”
+// “finally”
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:298]:
```suggestion
// “最终”
```

**Review by yin1999 — COMMENTED** [review]
一些建议

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md (hunk 1 above)
Comment (yin1999) [reply files/zh-cn/web/javascript/reference/statements/try...catch/index.md:10]: Last nit here :eyes: @JasonLamv-t

Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:10]:
```suggestion
**`try...catch`** 语句由一个 `try` 块和一个 `catch` 块或`finally` 块组成，或者两者皆有。首先执行 `try` 块中的代码，如果它抛出异常，则将执行 `catch` 块中的代码。`finally` 块中的代码将在控制流退出整个结构之前始终被执行。
```

Comment (JasonLamv-t) [reply files/zh-cn/web/javascript/reference/statements/try...catch/index.md]: I'm sorry for not getting back to you sooner. Quite busy this period.

File: files/zh-cn/web/javascript/reference/statements/try...catch/index.md [hunk 15]
```diff
@@ -1,118 +1,100 @@
...
 title: try...catch
 slug: Web/JavaScript/Reference/Statements/try...catch
+l10n:
+  sourceCommit: 4e947e81afc753bedcaaba75f262a07b92511849
 ---
 
 {{jsSidebar("Statements")}}
 
-**`try...catch`** 语句标记要尝试的语句块，并指定一个出现异常时抛出的响应。
+**`try...catch`** 语句由一个 `try` 块和一个 `catch` 块或`finally` 块组成，或者两者皆有。首先执行 `try` 块中的代码，如果它抛出异常，则将执行 `catch` 块中的代码。`finally` 块中的代码将在控制流退出整个结构之前始终被执行。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/try...catch/index.md:10]:
```suggestion
**`try...catch`** 语句由一个 `try` 块和一个 `catch` 块或 `finally` 块（或两者皆有）组成。首先执行 `try` 块中的代码，如果它抛出异常，则将执行 `catch` 块中的代码。`finally` 块中的代码将在控制流退出整个结构之前始终被执行。
```

**Review by yin1999 — APPROVED** [review]
👍

**Pull request merged**